backuprestore/restore.py

```diff
@@ -159,12 +159,8 @@ def main():
                 iaas_client.exit('Could not create the following directory: {}'.format(
                     '{}/blueprint/files'.format(DIRECTORY_PERSISTENT)))
             # +-> Copy the Encrypted backups contents to the persistent volume
-            if not iaas_client.copy_directory(
-                    '{}/blueprint/files/*'.format(DIRECTORY_DOWNLOADS),
-                    '{}/blueprint/files'.format(DIRECTORY_PERSISTENT)):
-                iaas_client.exit('Could not copy from {}/{} to the persistent volume.'
-                                 .format(DIRECTORY_DOWNLOADS, DIRECTORY_PERSISTENT))
-
+	          if not iaas_client.shell('cp -r {}/blueprint/files/* {}/blueprint/files; exit 0'.format(DIRECTORY_DOWNLOADS, DIRECTORY_PERSISTENT)):
```

Comment:
as discussed please check if the folder is empty first - 

if yes , exit 0
if no , 'cp -r {}/blueprint/files/* {}/blueprint/files;'

Reply:
+1

Reply:
use 

`ls -A <path>/blueprint/files/`

which ignore . and ..

ls --help
-A, --almost-all           do not list implied . and ..

Reply:
Do as follows: 

if not iaas_client.shell('ls -A /var/vcap/store/blueprint/files/').rstrip():
         skip copy
else:
         copy

Reply:
ok..
I tried with 'ls -A' and rsync and was somehow unable to make it work.
So, I used os.listdir() which returns empty array ([]) if the list is empty, i.e., for empty folders it will return [] and hence, for non-empty folders only, the copy operation will occur.
Hope this is fine.

Reply:
yes. if works ... go ahead

Reply:
+1 .

@pritishmishra  : please talk to @akskap and create a blueprint boshrelease and use the new version in sf product component.